src/Console/Command/MigrateCommand.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MX\PhinxMigrations\Console\Command;
+
+use Magento\Setup\Console\Command\DbStatusCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+/**
+ * @package MX\PhinxMigrations
+ * @author  James Halsall <james.halsall@inviqa.com>
+ */
+class MigrateCommand extends Command
+{
+    protected function configure()
+    {
+        $this
+            ->setName('mx:migrate')
+            ->setDescription('Migrates the database so it is ready for use with the current application code.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $dbStatusCommand = $this->getApplication()->get('setup:db:status');
+
+        $output->writeln('<info>Checking if setup:upgrade is required...</info>');
+        $exit = $dbStatusCommand->run(new ArrayInput([]), new NullOutput());
+
+        if ($exit === DbStatusCommand::EXIT_CODE_UPGRADE_REQUIRED) {
+            $output->writeln('<info>A setup:upgrade is required, running...</info>');
+            $setupUpgradeCommand = $this->getApplication()->get('setup:upgrade');
+            $setupUpgradeCommand->run(new ArrayInput([]), $output);
```

Comment:
Normally after a production deployment we need to run setup:upgrade with the "--keep-generated" option, because in production mode we compile the di and deploy it with the code to production which we don't want to loose ofc.

So this migrate command should be able to accept a "--keep-generated" option as well and pass it over to the setup:upgrade here.